### PR TITLE
 Added basic integration tests for cluster dns operator 

### PR DESCRIPTION
**Pull request by pravisankar**
```
KUBECONFIG=</path/to/admin.kubeconfig> make test-integration
```

### CODE REVIEW

**Comment by pravisankar** [comment]
@openshift/sig-network-edge PTAL

File: test/integration/integration_test.go [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+func getDNSPod(ns string) (string, error) {
+	var podName string
+
+	var err error
+	kubeClient := kubeclient.GetKubeClient()
+
+	err = wait.Poll(500*time.Millisecond, time.Minute, func() (bool, error) {
+		// NOTE: kubeClient is used instead of sdk.List(ns, podList) as the
+		// latter is not reliably populating pod.Status.Phase field at this time.
```
Comment (ironcladlou) [inline test/integration/integration_test.go]: Details? Sounds serious

File: test/integration/integration_test.go [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+		podList, err := kubeClient.Core().Pods(ns).List(metav1.ListOptions{})
```
Comment (ironcladlou) [inline test/integration/integration_test.go]: To make sure this isn't flaky, probably want to add in the DaemonSet selector or something. There might be some upstream util for this

File: test/integration/integration_test.go [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+			dnsType: "srv",
+		},
+		{
+			domain:  fmt.Sprintf("_http._tcp.%s.%s.svc.%s.", appService.Name, appService.Namespace, clusterDomain),
+			dnsType: "srv",
+		},
+	}
+
+	for _, test := range tests {
+		digCmd := fmt.Sprintf("dig +noall +answer %s %s @%s -p %s", test.dnsType, test.domain, serviceIP, servicePort)
```
Comment (ironcladlou) [inline test/integration/integration_test.go]: net.LookupHost would be more portable but then resolution becomes platform/environment dependent... not sure what the best approach is here. Fine with dig for now, just wanted to start thinking about it

**Comment by ironcladlou** [comment]
This shook out some annoying stuff with the operator SDK and a question about how to reliably test DNS given the peculiarities of resolver impls, but it's a great start.

/lgtm

**Pull request merged**